### DISCUSSION TITLE
Significant bugs with automatic discounts

### DISCUSSION THREAD

**Discussion by nboliver** [Shopify/storefront-api-feedback#106] [Help]
If a Buy X Get Y discount is set up with the same product for X and Y (a BOGO, if you will), the cart line items returned by the API are incorrect (the discount is applied to both items). The cart's estimated cost is correct, however. There are also issues related to incrementing line items that the discount applies to. Sometimes the quantity returned by the API is incorrect, or the discount is calculated incorrectly.

<img width="632" alt="image" src="https://user-images.githubusercontent.com/3094055/166804373-8568cce1-23f7-4b61-a9c7-055eb38023bc.png">

<img width="960" alt="image" src="https://user-images.githubusercontent.com/3094055/166804954-be5c7287-702d-4754-bca8-e06e429439a5.png">

Can provide API response if that's helpful.

**Comment by davidboya** [comment]
I have the same problem with automatic discount, trying to setup a percentage discount on 2nd product.
